packages/matchbox/src/components/Banner/Banner.js

```diff
@@ -1,144 +1,151 @@
-import React, { Component } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
-import classnames from 'classnames';
-import { Error as ErrorIcon, CheckCircle, InfoOutline, Close } from '@sparkpost/matchbox-icons';
-
-import styles from './Banner.module.scss';
+import { createPropTypes } from '@styled-system/prop-types';
+import { Close } from '@sparkpost/matchbox-icons';
+import { Box } from '../Box';
+import { Text } from '../Text';
+import { Inline } from '../Inline';
+import { ScreenReaderOnly } from '../ScreenReaderOnly';
+import styled from 'styled-components';
+import { container, childLinks, statusIcons, dismissBase, dismissColor } from './styles';
+import { buttonReset } from '../../styles/helpers';
 import { buttonFrom } from '../Button';
+import { margin } from 'styled-system';
 
-const actionOverrides = { outline: true, size: 'small' };
-
-const IconSection = ({ status }) => {
-  const icons = {
-    success: {
-      iconName: CheckCircle,
-      iconLabel: 'Success'
-    },
-    info: {
-      iconName: InfoOutline,
-      iconLabel: 'Info'
-    },
-    warning: {
-      iconName: ErrorIcon,
-      iconLabel: 'Warning'
-    },
-    danger: {
-      iconName: ErrorIcon,
-      iconLabel: 'Error'
-    }
-  };
-
-  if (status === 'default' || !icons[status]) {
-    return null;
-  }
-
-  const iconClasses = classnames(
-    styles.Icon,
-    status && styles[`${status}Icon`]
-  );
+function IconSection({ status }) {
+  const statusIcon = React.useMemo(() => {
+    return status === 'default' ? statusIcons.info : statusIcons[status];
+  }, [statusIcons, status]);
 
-  const Icon = icons[status].iconName;
-  const iconLabel = icons[status].iconLabel;
+  const Icon = statusIcon.iconName;
 
   return (
-    <div className={styles.IconWrapper}>
-      <Icon size={30} className={iconClasses} label={iconLabel}/>
-      <div className={styles.IconBackdrop} />
-    </div>
+    <Box
+      position="relative"
+      // IconSection do not render under the first breakpoint
+      display={['none', 'none', 'flex']}
```

Comment:
That works well. `<Box/>` to the rescue.

The only issue is actually an a11y problem with doing that...technically a user with color blindness won't be able to see what the status of a `<Banner/>` is without the icon there as well on all viewports.

Any other possible routes to take? I have seen some design systems move the icon above the rest of the content, or possibly make it substantially smaller on more narrow viewports.

Reply:
Oh right, good call. I'll probably add the icon back in and align it to the top left of the Banner